tracee/event_monitor_ebpf.c

```diff
@@ -1145,6 +1166,12 @@ static __always_inline int trace_ret_generic(struct pt_regs *ctx, u32 id, u64 ty
     if (!should_trace())
         return -1;
 
+    // Workaround to a bug in gobpf, where a kprobe event (e.g. mmap syscall) can't be attached to two different functions.
+    // As we need to use save_args in kprobe entry, but one of the events (functions) might be chosen and the other not,
+    // We need a mechanism that can tell if an event was chosen by the user, so we can save the args without enabling both events.
+    if (!event_chosen(id))
+        return 0;
+
```

Comment:
can you please also open an issue in tracee to explain this workaround as a reminder for us to remove this in the future? I think you will explain it better than me

Reply:
Sure. Will do.

Reply:
Actually when thinking about it - this workaround is actually not a bad thing if it will stay.
It is a good idea to have a mechanism that sends the event only if the user requested for it.
Maybe we should just remove the "workaround" comment? WDYT?